src/components/fields/date-range-field/date-range-field.tsx

```diff
@@ -47,10 +48,12 @@ export const DateRangeField = (props: IGenericFieldProps<TDateRangeFieldSchema>)
 		const maxDateRule = validation?.find((rule) => "maxDate" in rule);
 		const isRequiredRule = validation?.find((rule) => "required" in rule);
 		const excludedDatesRule = validation?.find((rule) => "excludedDates" in rule);
+		const noOfDaysRule = validation?.find((rule) => "numberOfDays" in rule);
 
 		const minDate = DateTimeHelper.toLocalDateOrTime(minDateRule?.["minDate"], dateFormat, "date");
 		const maxDate = DateTimeHelper.toLocalDateOrTime(maxDateRule?.["maxDate"], dateFormat, "date");
 
+		setNoOfDays(noOfDaysRule?.["numberOfDays"]);
```

Comment:
sorry missed out that this `useEffect` already has logic to set the `derivedProps` state. can combine `noOfDays` with `derivedProps` instead